### ISSUE TITLE
KnowCode Example with custom layout.xml not working 

### ISSUE THREAD

**Issue by magicallyindia** [TotalCross/totalcross#51]
## Issue Title
We have created a sample with the KnowCode Sample using android XML File . After running the code we are getting the below error 

## Describe the bug
#0 - the file given: tcapp.prop
#1 - dataPath: tcapp.prop
#2 - MainWindow's path from current folder: in/conceptive/test1/tcapp.prop
#3 - classpath
#4 - jar file
#4b - C:/Users/bhush/.m2/repository/com/totalcross/totalcross-sdk/6.0.4/totalcross-sdk-6.0.4.jar
file not found

java.lang.Exception: Layout do xml não consta lista dos conhecidos.
        at com.totalcross.knowcode.parse.XmlContainerFactory.create(XmlContainerFactory.java:50)
        at com.totalcross.knowcode.parse.XmlContainerFactory.create(XmlContainerFactory.java:74)
        at in.conceptive.test1.HelloMyEmbeddedWorld.initUI(HelloMyEmbeddedWorld.java:22)
        at totalcross.ui.MainWindow.startProgram(MainWindow.java:515)
        at totalcross.ui.MainWindow._onTimerTick(MainWindow.java:556)
        at totalcross.Launcher$WinTimer$1.run(Launcher.java:383)
        at totalcross.TCEventThread.privatePumpEvents(TCEventThread.java:70)
        at totalcross.TCEventThread.run(TCEventThread.java:44)
---------------------------
>>>>>>> CAUGHT UNHANDLED EXCEPTION IN EVENT THREAD:
java.lang.NullPointerException
        at totalcross.ui.Container.add(Container.java:288)
        at totalcross.ui.Window.swap(Window.java:1592)
        at in.conceptive.test1.HelloMyEmbeddedWorld.initUI(HelloMyEmbeddedWorld.java:23)
        at totalcross.ui.MainWindow.startProgram(MainWindow.java:515)
        at totalcross.ui.MainWindow._onTimerTick(MainWindow.java:556)
        at totalcross.Launcher$WinTimer$1.run(Launcher.java:383)
        at totalcross.TCEventThread.privatePumpEvents(TCEventThread.java:70)
        at totalcross.TCEventThread.run(TCEventThread.java:44)

## Additional context
Attached is the sample code we have done in VS Code 
TotalCross 6.0.4 , KnowCodeXML latest branch from git repository
[sample12.zip](https://github.com/TotalCross/totalcross/files/4770576/sample12.zip)

We are seriously evaluating this for a realtime project and need documentation / issue list so that this can be adapted shortly in production. 

Thanks and regards 

Bhushan 
CEO - Conceptive

**Comment by brunoamuniz** [comment]
Hey @magicallyindia, thanks for reaching out! =)

KnowCode is fully working with develop version of TotalCross, right now. We will release the version 6.1.0 of TotalCross until the end of June.

If you want to try out KnowCode, you can compile develop branch or also, I am attaching the compiled version here https://www.dropbox.com/s/biytdnwafvywk54/totalcross_deploy.zip?dl=0 (we are working to provide nightly builds to make this easier).

You can get more info about compiling TotalCross here:
https://www.youtube.com/watch?v=eHlgsB3f4Pw
https://learn.totalcross.com/documentation/developers-area/how-to-build-totalcross-vm-wip#linux-arm
https://learn.totalcross.com/documentation/developers-area/how-to-build-totalcross-sdk

We are also trying to run your sample app, we will give you a feedback soon.

Cheers,

**Comment by magicallyindia** [comment]
Thanks ,i will try the nightly build and also build totalcross. Will come back with any issues. Please let me know if i could send an email. We are evaluating this for an important device due to time constraint .

**Comment by magicallyindia** [comment]
I have found one bug in the NodeSax.java method setWp() line number 947 

Previously it was  System.out.println(wp);  -> This raises error for printstream 

I just appended  System.out.println(" "+wp); -> This is right way so it fixed it. The double value must be concatenated with string.

**Comment by magicallyindia** [comment]
> Hey @magicallyindia, thanks for reaching out! =)
> 
> KnowCode is fully working with develop version of TotalCross, right now. We will release the version 6.1.0 of TotalCross until the end of June.
> 
> If you want to try out KnowCode, you can compile develop branch or also, I am attaching the compiled version here https://www.dropbox.com/s/biytdnwafvywk54/totalcross_deploy.zip?dl=0 (we are working to provide nightly builds to make this easier).
> 
> You can get more info about compiling TotalCross here:
> https://www.youtube.com/watch?v=eHlgsB3f4Pw
> https://learn.totalcross.com/documentation/developers-area/how-to-build-totalcross-vm-wip#linux-arm
> https://learn.totalcross.com/documentation/developers-area/how-to-build-totalcross-sdk
> 
> We are also trying to run your sample app, we will give you a feedback soon.
> 
> Cheers,

With the new NightlyBuild its working , I'm able to load the XML from android studio. Will test and let you know on Embedded Linux. 

Bhushan

**Comment by brunoamuniz** [comment]
hey @magicallyindia you can reach me out at bruno@totalcross.com too

Cheers,

**Comment by magicallyindia** [comment]
> hey @magicallyindia you can reach me out at [bruno@totalcross.com](mailto:bruno@totalcross.com) too
> 
> Cheers,

Thanks

**Comment by acmlira** [comment]
Hey guys, we released version 6.1.0 (#89) with some delay (sorry 😢), but I think that now this issue may be solved 😅

**Comment by ItaloYeltsin** [comment]
Hello @magicallyindia ,

in order to solve this problem, update the following parts of your `pom.xml`.

your totalcross-sdk version java dependency:
```xml
<dependency>
            <groupId>com.totalcross</groupId>
            <artifactId>totalcross-sdk</artifactId>
            <version>6.1.0</version> <!-- This line: -->
</dependency>
```

the totalcross-maven-plugin:
```xml
<plugin>
  <groupId>com.totalcross</groupId>
  <artifactId>totalcross-maven-plugin</artifactId>
  <version>1.1.6</version> <!-- This line: -->
  <configuration>
    <name>${project.name}</name>
    <platforms>
```
